pageserver/src/disk_usage_eviction_task.rs

```diff
@@ -469,106 +443,160 @@ pub async fn disk_usage_eviction_task_iteration_impl<U: Usage>(
     }))
 }
 
-/// Different modes of gathering tenant's least recently used layers.
-#[derive(Debug)]
-enum Mode {
-    /// Add all but the most recently used `min_resident_size` worth of layers to the candidates
-    /// list.
-    ///
-    /// `min_resident_size` defaults to maximum layer file size of the tenant. This ensures that
-    /// the tenant will always have one layer resident. If we cannot compute `min_resident_size`
-    /// accurately because metadata is missing we use hardcoded constant. `min_resident_size` can
-    /// be overridden per tenant for important tenants.
-    RespectTenantMinResidentSize,
-    /// Consider all layer files from all tenants in LRU order.
-    ///
-    /// This is done if the `min_resident_size` respecting does not relieve pressure.
-    GlobalLru,
+#[derive(Clone)]
+struct EvictionCandidate {
+    timeline: Arc<Timeline>,
+    layer: Arc<dyn PersistentLayer>,
+    last_activity_ts: SystemTime,
 }
 
-/// Figure out eviction candidates for the given tenant and append them to `lru_candidates`.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+enum MinResidentSizePartition {
+    Above,
+    Below,
+}
+
+enum EvictionCandidates {
+    Cancelled,
+    Finished(Vec<(MinResidentSizePartition, EvictionCandidate)>),
```

Comment:
Will the `MinResidentSizePartition`'s be the same for many items before changing and remaining the same or is the distribution more like random?

Reply:
Ok, I'm thinking if `usize, Vec<EvictionCandidate>` would be more $insert_adjective, if this is clear cut as it was in https://github.com/neondatabase/neon/commit/386c2d0112ddecff45f4c49920e4a4a51e9a60c7, but lets not do this premature optimization.